R/plotDataSetList.R

```diff
@@ -1173,25 +1174,26 @@ plot_general_data <- function(df, x_attr = 'algId', y_attr = 'vals', type = 'vio
                }
              }
              
+          
              suppressWarnings(
                p %<>%
                  add_trace(
                    data = df, x = ~x, y = ~y, color = ~l, legendgroup = ~l,
                    type = 'scatter', mode = 'lines+markers', color = ~l,
                    linetype = ~l, marker = list(size = getOption('IOHanalyzer.markersize', 4)), linetypes = dashes,
                    colors = colors, showlegend = show.legend,
-                   text = y_attr, line = list(width = getOption('IOHanalyzer.linewidth', 2)),
+                   text = ~text, line = list(width = getOption('IOHanalyzer.linewidth', 2)),
```

Comment:
Just a quick question: did you observe if this way of plotting (using formula color = `~l`) is faster than the old ways, i.e., looping over the grouping variable?

Reply:
Yes, this should be faster than just looping, as data.tables are quite efficient for these type of operations.